### PR TITLE
Union optmisation

### PR DESCRIPTION
**Pull request by WirajanDASILVA**
* **Please check if the PR fulfills these requirements**
- [X] The commit messages follows our guidelines (explicit, in english)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [X] Bug fix:
- [ ] New features:
- [ ] Documentation
- [X] Optimization

* **Other information**:
Set Operations methods like union and subtraction methods have been optimised. it no more merges the resulting faces automatically. If merging adjacent faces is required to have a cleaner object, it must be done post union operation.

### CODE REVIEW

File: volmdlr/faces.py [hunk 1]
```diff
@@ -3712,13 +3712,26 @@ def inner_contours3d(self):
...
         this error is raised to enforce overloading of this method
         """
         raise NotImplementedError(
-            '_bounding_box method must be overloaded by {}'.format(
-                self.__class__.__name__))
+            f"bounding_box method must be"
+            f"overloaded by {self.__class__.__name__}")
+
+    @bounding_box.setter
+    def bounding_box(self, new_bounfing_box):
```
Comment (masfaraud) [inline volmdlr/faces.py:3725]: typo: bounfing

File: volmdlr/wires.py [hunk 2]
```diff
@@ -2720,76 +2832,166 @@ def repositioned_polygon(self, x, y):
...
+                    else:
+                        closing_point = intr_list[1].end
+            elif intr_list[0] == middle_point and \
+                    polygon_primitive.length() == intr_list[1].length():
+                closing_point = intr_list[1].start
+                distance = 0
+
+        return closing_point
+
+    def search_farthest(self, point, possible_closing_points):
```
Comment (masfaraud) [inline volmdlr/wires.py:2919]: docstring

**Review by masfaraud — CHANGES_REQUESTED** [review]

File: volmdlr/wires.py (hunk 2 above)
Comment (WirajanDASILVA) [reply volmdlr/wires.py:2919]: It's been added in [this PR](https://github.com/Dessia-tech/volmdlr/pull/332)

File: volmdlr/wires.py [hunk 3]
```diff
@@ -4212,20 +4372,30 @@ def validate_closing_point(closing_point_index, list_closing_point_indexes,
...
+                    else:
+                        break
+                # list_remove_closing_points.extend(
+                #     [list_closing_point_indexes[-1],
+                #      list_closing_point_indexes[-2],
+                #      list_closing_point_indexes[-3],
+                #      list_closing_point_indexes[-4]])
+                # print('list_remove_closing_points source_2:',
+                #       list_remove_closing_points)
+                # print('source closing_point:', closing_point_index)
```
[the remark this answers is not in the document]
Comment (WirajanDASILVA) [reply volmdlr/wires.py:4398]: It's been cleaned in [this PR](https://github.com/Dessia-tech/volmdlr/pull/332)

**Review by masfaraud — APPROVED** [review]

**Pull request merged**